config/config.go

```diff
@@ -33,20 +34,25 @@ func (conf *TerragruntConfig) String() string {
 
 // SubstituteAllVariables replace all remaining variables by the value
 func (conf *TerragruntConfig) SubstituteAllVariables(terragruntOptions *options.TerragruntOptions) {
-	substitute := func(value *string) string {
+	substitute := func(value *string) *string {
+		if value == nil {
+			return nil
+		}
+
 		*value = SubstituteVars(*value, terragruntOptions)
-		return *value
+		return value
 	}
 
 	substitute(&conf.Uniqueness)
+	substitute(conf.AssumeRole)
 	if conf.Terraform != nil {
 		substitute(&conf.Terraform.Source)
 	}
 	if conf.RemoteState != nil && conf.RemoteState.Config != nil {
 		for key, value := range conf.RemoteState.Config {
 			switch val := value.(type) {
 			case string:
-				conf.RemoteState.Config[key] = substitute(&val)
+				conf.RemoteState.Config[key] = *substitute(&val)
```

Comment:
Dereference a pointer. The function returns a pointer to string and we must dereference it to get the string.